packages/wrangler/src/user/user.ts

```diff
@@ -1249,3 +1252,27 @@ async function fetchAuthToken(body: URLSearchParams) {
 		headers,
 	});
 }
+
+async function getJSONFromResponse(response: Response) {
+	const text = await response.text();
```

Comment:
Very minor, but if you passed in the text string to this function, it would reduce its responsibilities:

```
async function getJSONFromText<T>(text: string): T {
  try ...
}

...

const json = getJSONFromText<TokenResponse>(await response.text());
```

Reply:
RIght, but then we would have to extract the text on each of the three calls, which seems a bit repetitive; but more importantly, the logs grab the status from the response so that would need to be passed in too.

Reply:
Also, we might also want to use the status code as part of the analysis of what went wrong with the response (e.g. I think that the Bot Management failure would have a 403 status).